react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,48 @@
+import { ImageCard } from './ImageCard.tsx';
+import { NextButton } from './NextButton.tsx';
+import { PrevButton } from './PrevButton.tsx';
+import { Dots } from './Dots.tsx';
+import { useState, useEffect } from 'react';
+
+export type Image = {
+  src: string;
+  alt: string;
+};
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setCurrentIndex((currentIndex + 1) % images.length);
+    }, 3000);
+    return () => clearTimeout(timerId);
+  }, [currentIndex, images.length]);
+
+  function handleNextImage() {
+    setCurrentIndex((prevIndex) => (prevIndex + 1) % images.length);
```

Comment:
In these ones, you can actually just use the simpler `setCurrentIndex((currentIndex + 1) % images.length);`.

I usually use this simpler form first, and then only switch to the more complex form with an arrow function only if problems arise because of closures.